core/src/main/java/com/linecorp/armeria/common/DefaultHttpHeaders.java

```diff
@@ -186,16 +186,16 @@ public HttpHeaders status(HttpStatus status) {
     @Nullable
     @Override
     public MediaType contentType() {
-        final MediaType contentType = this.contentType;
-        if (contentType != null) {
-            return contentType;
-        }
-
         final String contentTypeString = get(HttpHeaderNames.CONTENT_TYPE);
         if (contentTypeString == null) {
             return null;
         }
 
+        final MediaType contentType = this.contentType;
+        if (contentType != null && contentType.toString().equals(contentTypeString)) {
```

Comment:
I believe this will always be `false` if the content type header was formatted weird, with extra spaces for example since `toString` is always normalized. It's probably rare enough to not worry about, but what do you think? I guess we could fix this without the efficiency loss of the map lookup if we overrode the logic of the setter instead of the getter

Reply:
If we override the setter, we would need to check if the header name is equal to 'content-type' for all headers. I guess this would be more inefficient?

Reply:
There are three ways to set a content type which are `contentType(mediaType)`, `set*()` and `add*()`. They are all ended up calling `set(AsciiString name, String value)`. If we override the `set` method, we should check the name equals to `HttpHeaderNames.CONTENT_TYPE` every time add or set header.

Reply:
There are three ways to set a content type which are `contentType(mediaType)`, `set*()` and `add*()`. They are all ended up calling `set(AsciiString name, String value)`. If we override the `set` method, we should check the name equals to `HttpHeaderNames.CONTENT_TYPE` every time add or set header.